packages/harpguru-core/src/components/harp-cell/harp-cell.tsx

```diff
@@ -50,24 +51,35 @@ export const HarpCell = ({ yxCoord }: HarpCellProps): React.ReactElement => {
   const handleTapStateChange = ({
     nativeEvent,
   }: TapGestureHandlerStateChangeEvent) => {
-    if (nativeEvent.state !== State.END) return
-
-    addBufferedActivityToggle(thisDegreeId)
+    const interestingTouchStates = [State.BEGAN, State.CANCELLED, State.FAILED]
```

Comment:
Shouldn't be called "interesting..." because the `UNDETERMINED` is also of interest, it's just not not a direct driver of it's own state update. That is left to the `useEffect` driven by `isActive`.

If we could get this function from the same place that we set the `useEffect` then it would be as clear as possible that they are related. Seems like a manageable custom hook.

Reply:
Done the renaming

Reply:
Created custom hook with these two bits of logic brought together